tilings/algorithms/minimal_gridded_perms.py

```diff
@@ -395,27 +395,26 @@ def minimal_gridded_perms(
                 yield GriddedPerm.empty_perm()
             return
         if len(self.requirements) == 1:
-            for req in self.requirements[0]:
-                yield req
+            yield from self.requirements[0]
             return
 
         # a priority queue sorted by length of the gridded perm. This is
         # ensured by overwriting the __lt__ operator in the Info class.
         queue: List[QueuePacket] = []
         heapify(queue)
 
-        initial_gps_to_auto_yield: Set[GriddedPerm] = set()
+        initial_gps_to_auto_yield: Dict[int, Set[GriddedPerm]] = defaultdict(set)
         yielded: Set[GriddedPerm] = set()
         for gp in self._prepare_queue(queue):
             if yield_non_minimal:
                 yielded.add(gp)
                 yield gp
             else:
-                initial_gps_to_auto_yield.add(gp)
+                initial_gps_to_auto_yield[len(gp)].add(gp)
 
         def yielded_subgridded_perm(gp: GriddedPerm) -> bool:
             """Return True if a subgridded perm was yielded."""
-            return gp.contains(*yielded, *initial_gps_to_auto_yield)
+            return gp.contains(*yielded, *initial_gps_to_auto_yield[len(gp)])
```

Comment:
you need to check for all sizes in range(curr_len + 1, len(gp) + 1)

Reply:
Actually I think I only need to check for containment in the yielded gridded perms. It's what I did and it seems to work

Reply:
I think you don't need check anything in the initial in fact. If they've already been yielded then they are in the yielded set and the function already returns true. At least I did it that way and it works.

Reply:
This is a good observation - they are all added to the yielded set the moment the first one of that length is found. Hence for gp, yielded contains all initial_gps of length <= len(gp)

Reply:
I don't see it as a problem it will just yield the one point extension and then when it changes length it won't yield the one from `initial_gp_to_auto_yield` since it is already in yielded.

Also, I'm not sure I understand everything about this 1 point extension but I'm not sure I understand how it's not breaking the fact that the permutation are yielded in increasing length?

Reply:
You are right, the order is preserved as the point extension would be yielded first, good point!

I am going to try abstract a bit to make the second question super clear:
- the prepare queue finds some gps, call it S that we know satisfy the obs and reqs
- we pop a gridded perm gp from the queue, 
- we know at this point that all from S with length <= len(gp) are yielded. 
- we then try yield the one point extensions of gp, each of length len(gp) + 1. 
- and repeat

If a point extension isn't a minimal gridded perm, it is added to the priority queue (sorted by length), this is why we ensure they are yielded in increasing length.

Reply:
Ok I think I understand. Any gp on the queue does not satisfy the requirements. Hence, when `len(qpacket.gp) == n`, we are actually yielding permutation of length n+1.

Am I right to think that we should therefore yield the initial_to_auto_yield first by changing line 493-496 from
```python
             initial_to_yield = chain.from_iterable(
                if not yielded_subgridded_perm(qpacket.gp):
                    initial_gps_to_auto_yield.pop(i, tuple())
                    yielded.add(qpacket.gp)
                    for i in range(curr_len + 1, new_len + 1)
                    yield qpacket.gp
                )
```
to
```python
             initial_to_yield = chain.from_iterable(
                if not yielded_subgridded_perm(qpacket.gp):
                    initial_gps_to_auto_yield.pop(i, tuple())
                    yielded.add(qpacket.gp)
                    for i in range(curr_len + 1, new_len + 2)
                    yield qpacket.gp
                )
```

Reply:
I tested it locally and it passes all the tests

Reply:
I agree with this change